connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/ProtobufDataToCatalyst.scala

```diff
@@ -108,18 +112,16 @@ private[protobuf] case class ProtobufDataToCatalyst(
     val binary = input.asInstanceOf[Array[Byte]]
     try {
       result = DynamicMessage.parseFrom(messageDescriptor, binary)
```

Comment:
Functionally there shouldn't be any difference. Likely benefit is better performance. We don't support other feature like extensions.
Since Java class is likely less used than other methods, perf gain may not be that useful. I will leave comment.

Reply:
Added a comment about it.